pydis_site/apps/api/tests/test_infractions.py

```diff
@@ -305,6 +312,167 @@ def test_returns_400_for_non_hidden_required_hidden_type(self):
             'hidden': [f'{data["type"]} infractions must be hidden.']
         })
 
+    def test_returns_400_for_active_infraction_of_type_that_cannot_be_active(self):
+        """Test if the API rejects active infractions for types that cannot be active."""
+        url = reverse('bot:infraction-list', host='api')
+        restricted_types = ('note', 'warning', 'kick')
+
+        for infraction_type in restricted_types:
+            with self.subTest(infraction_type=infraction_type):
+                invalid_infraction = {
+                    'user': self.user.id,
+                    'actor': self.user.id,
+                    'type': infraction_type,
+                    'reason': 'Take me on!',
+                    'hidden': True,
+                    'active': True,
+                    'expires_at': '2019-10-04T12:52:00+00:00'
+                }
+                response = self.client.post(url, data=invalid_infraction)
+                self.assertEqual(response.status_code, 400)
+                self.assertEqual(response.json(), {
```

Comment:
I don't believe this is consistent with our code style but honestly it currently looks better than what it's "supposed" to be:

```py
self.assertEqual(
    response.json(), 
    {
        'active': [f'{infraction_type} infractions cannot be active.']
    }
)
```

It wouldn't be as bad if the dict was on one line:

```py
self.assertEqual(
    response.json(), 
    {'active': [f'{infraction_type} infractions cannot be active.']}
)
```

Reply:
Oh, so I'm not sure if this should be changed or not. Thoughts?

Reply:
I'll change that.